src/date/format-properties.js

```diff
@@ -208,6 +209,58 @@ return function( pattern, cldr ) {
 
 			// Zone
 			case "z":
+
+				//z...zzz: "{shortRegion}", eg. "PST" or "PDT".
+				//zzzz: "{regionName} {Standard Time}" or "{regionName} {Daylight Time}",
+				//eg. "Pacific Standard Time" or "Pacific Daylight Time".
+				if ( properties.timeZone ) {
+
+					// The latest metazone data of the metazone array.
+					//TODO expand to support the historic metazones based on the given date.
+					var standardTzName,
+						daylightTzName,
+						metaZone = cldr.supplemental([
+							"metaZones/metazoneInfo/timezone", properties.timeZone, 0,
+							"usesMetazone/_mzone"
+						]);
+
+					if ( metaZone ) {
+						standardTzName = cldr.main([
+							"dates/timeZoneNames/metazone",
+							metaZone,
+							length < 4 ? "short" : "long",
+							"standard"
+						]);
+						daylightTzName = cldr.main([
+							"dates/timeZoneNames/metazone",
+							metaZone,
+							length < 4 ? "short" : "long",
+							"daylight"
+						]);
+						if ( !standardTzName && !daylightTzName ) {
```

Comment:
this will help on the scenario like https://github.com/globalizejs/globalize/pull/687/files#diff-df6075946ae91cbb093f63c7597cd746R460 where the timezone id is valid not supported in CLDR.